### PR TITLE
Whitelist apps

### PR DESCRIPTION
**Pull request by gmgigi96**
This PR introduces the following enhancements:
1. Overrides the AppProvider supported mimetypes from the configuration
2. Add a configuration for the mime types in the AppRegistry (a friendly name, default AppProvider, description, icon, ...)

### CODE REVIEW

**Review by glpatcern — COMMENTED** [review]
Looks all good, I just want to prepare a good list of default mime types to be included. Will do that on Monday ;-)

File: internal/grpc/services/appregistry/appregistry_test.go [hunk 1]
```diff
@@ -168,6 +186,45 @@ func Test_GetAppProviders(t *testing.T) {
...
+			"icon":        "https://example.org/icons&file=json.png",
+			"default_app": "some Address",
+		},
+		"text/xml": {
+			"extension":   "xml",
+			"name":        "XML File",
+			"icon":        "https://example.org/icons&file=xml.png",
+			"default_app": "some Address",
+		},
+		"application/vnd.openxmlformats-officedocument.wordprocessingml.document": {
```
Comment (glpatcern) [inline internal/grpc/services/appregistry/appregistry_test.go:202]: I propose to keep the test examples smaller and remove all except JSON and XML. Then we should populate the example configuration file with some sensible defaults.

**Review by glpatcern — CHANGES_REQUESTED** [review]
As quickly discussed, for the example config file I'd include the mime types for: docx, xlsx, pptx, odt, ods, odp, txt, md, root, ipynb.

File: internal/grpc/services/appregistry/appregistry_test.go (hunk 1 above)
Comment (gmgigi96) [reply internal/grpc/services/appregistry/appregistry_test.go:202]: As discussed, the mime types in the test are the minimum necessary to pass the tests.
For the example config, I've added all the extensions you mentioned except the root as it does not have a defined mime type.

**Review by glpatcern — APPROVED** [review]
All good for me!

**Pull request merged**